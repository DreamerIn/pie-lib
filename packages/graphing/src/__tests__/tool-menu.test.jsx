import { shallow } from 'enzyme';
import React from 'react';

import ToolMenu from '../tool-menu';

describe('ToolMenu', () => {
  let w;
  let onChange = jest.fn();
  const tools = [{ type: 'one' }, { type: 'two' }];

  const wrapper = extras => {
    const defaults = {
      classes: {},
      className: 'className',
      onChange,
      currentTool: tools[0],
      tools
    };
    const props = { ...defaults, ...extras };
    return shallow(<ToolMenu {...props} />);
  };
  describe('snapshot', () => {
    it('renders', () => {
      w = wrapper();
      expect(w).toMatchSnapshot();
    });
  });
  describe('logic', () => {
    describe('changeTool', () => {
      it('calls onChange', () => {
        w = wrapper();
        w.instance().changeTool('two');
        expect(onChange).toHaveBeenCalledWith(tools[1]);
      });
    });
  });
});
