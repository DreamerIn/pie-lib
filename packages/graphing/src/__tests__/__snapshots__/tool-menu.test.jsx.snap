// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToolMenu snapshot renders 1`] = `
<div
  className="className"
>
  <WithStyles(ToggleBar)
    onChange={[MockFunction]}
    options={Array []}
  />
</div>
`;
